### PR TITLE
feat: add array functions

### PR DESCRIPTION
**Pull request by mesejo**
- flatten
- range
- array_distinct
- unnest
- array_slice

### CODE REVIEW

File: python/letsql/tests/test_array.py [hunk 1]
```diff
@@ -112,3 +126,70 @@ def test_array_remove(con):
...
+            id="not_null",
+        ),
+    ],
+)
+def test_array_unique(con, data, expected):
+    t = ibis.memtable(data)
+    expr = t.a.unique()
+    result = con.execute(expr).map(frozenset, na_action="ignore")
+    expected = pd.Series(expected, dtype="object").map(frozenset, na_action="ignore")
+    assert frozenset(result.values) == frozenset(expected.values)
```
Comment (dlovell) [inline python/letsql/tests/test_array.py:195]: why do we use frozenset here?
could we test against the `pd.Series` result of `execute` directly?
```
In [20]: expr.execute().equals(pd.Series([[1, 3], [], [42], []]))
Out[20]: True
```

Comment (dlovell) [reply python/letsql/tests/test_array.py:195]: why does `expected` contain `float`s rather than `int`s?

File: python/letsql/tests/test_array.py [hunk 2]
```diff
@@ -112,3 +126,70 @@ def test_array_remove(con):
...
+def test_array_flatten(con, flatten_data, column, expected):
+    data = flatten_data[column]
+    t = ibis.memtable({column: data["data"]}, schema={column: data["type"]})
+    expr = t[column].flatten()
+    result = con.execute(expr)
+    assert_series_equal(
+        result.sort_values().reset_index(drop=True),
+        expected.sort_values().reset_index(drop=True),
+        check_names=False,
+    )
```
Comment (dlovell) [inline python/letsql/tests/test_array.py:149]: could this use `pd.Series.equals` as well?
```
expr.execute().equals(pd.Series([["abc"], ["bcd"], ["def"]]))
```

File: python/letsql/tests/test_array.py (hunk 1 above)
Comment (mesejo) [reply python/letsql/tests/test_array.py]: Fixed: I'm using `assert_series_equal`

**Review by dlovell — APPROVED** [review]

**Pull request merged**